libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridGenerateCircularGrid.cpp

```diff
@@ -0,0 +1,233 @@
+        double alfn = std::pow(parameters.maximum_uniform_size_rows, 1.0 / static_cast<double>(ninc));
+        double yValue = -blockSize * std::pow(alfn, static_cast<double>(1 - nuni));
+
+        for (int n = 0; n < static_cast<int>(yValues.size()); ++n)
+        {
+            yValue += (n < nuni ? blockSize : blockSize * std::pow(alfn, static_cast<double>(n + 1 - nuni)));
+            yValues[static_cast<UInt>(n)] = yValue;
+        }
+    }
+
+    return yValues;
+}
+
+lin_alg::Matrix<meshkernel::Point> meshkernel::CurvilinearGridGenerateCircularGrid::GenerateRectangularGrid(const MakeGridParameters& parameters)
+{
+    lin_alg::Matrix<Point> gridPoints(parameters.num_rows + 1, parameters.num_columns + 1);
+
+    const double x0 = parameters.origin_x;
+    const double y0 = parameters.origin_y;
+
+    const double phi = parameters.angle;
+    const double cs = std::cos(phi * constants::conversion::degToRad);
+    const double sn = std::sin(phi * constants::conversion::degToRad);
+
+    std::vector<double> xValues(ComputeXValues(parameters));
+    std::vector<double> yValues(ComputeYValues(parameters));
+
+    for (int n = 0; n < parameters.num_rows + 1; ++n)
+    {
+        for (int m = 0; m < parameters.num_columns + 1; ++m)
+        {
+            gridPoints(n, m) = Point(x0 + xValues[m] * cs - yValues[n] * sn, y0 + xValues[m] * sn + yValues[n] * cs);
+        }
+    }
+
+    return gridPoints;
+}
+
+std::vector<double> meshkernel::CurvilinearGridGenerateCircularGrid::ComputeRadiusValues(const MakeGridParameters& parameters)
+{
+    const UInt size = static_cast<UInt>(parameters.num_rows + 1);
+    const double blockSize = parameters.block_size_y;
+    const double r0 = parameters.radius_curvature;
+
+    std::vector<double> radiusValues(size);
+
+    if (parameters.uniform_rows_fraction == 1.0 || parameters.maximum_uniform_size_rows == 1.0)
+    {
+        double radius = r0;
+
+        for (UInt n = 0; n < size; ++n)
+        {
+            radiusValues[n] = radius;
+            radius += blockSize;
+        }
+    }
+    else
+    {
+        double radius = r0;
+
+        UInt nuni = static_cast<UInt>(static_cast<double>(size) * parameters.uniform_rows_fraction);
+        nuni += (size - 1 - nuni) % 2;
+        UInt ninc = (size - nuni) / 2 + 1;
+
+        double alfn = std::pow(parameters.maximum_uniform_size_rows, 1.0 / static_cast<double>(ninc));
+
+        for (UInt n = 0; n < size; ++n)
+        {
+            radius += (n < nuni ? blockSize : blockSize * std::pow(alfn, static_cast<double>(n - nuni)));
+            radiusValues[n] = radius;
+        }
+    }
+
+    return radiusValues;
+}
+
+std::vector<double> meshkernel::CurvilinearGridGenerateCircularGrid::ComputeThetaValues(const MakeGridParameters& parameters)
+{
+    const UInt size = static_cast<UInt>(parameters.num_columns + 1);
+    std::vector<double> thetaValues(size);
+
+    double deltaTheta = parameters.block_size_x / parameters.radius_curvature;
```

Comment:
const

Reply:
Neither thetaValues nor deltaTheta can be const